### DISCUSSION TITLE
1.7.0

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#826] [Обновления]
## Список изменений

- Переработана логика обновления переводов для меню расширения
- Добавлена кнопка для ручной проверки обновлений локализации
- Добавлена возможность копировавания текста из информации о расширении
- Добавлена настройка горячей клавиши перевода (#485 в #760)
- Добавлена поддержка путей `/inbox/` и `/bk/` для Видео Mail.ru (#763)
- Добавлена поддержка 9animetv с возможностью отображения субтитров от сайта (#748)
- Добавлена поддержка EpicGames Developers с возможностью отображения субтитров от сайта (#255, #505)
- Добавлена поддержка Odysee (#755)
- Добавлена поддержка Watchpornto
- Добавлена поддержка прямых ссылок на `.webm` файлы
- Добавлена поддержка Sap Learning с возможностью отображения субтитров от сайта (#719, #806)
- Добавлена поддержка Linkedin Learning с возможностью отображения субтитров от сайта (#332, [#372](https://github.com/ilyhalight/voice-over-translation/issues/372#issuecomment-1770306936))
- Добавлена возможность использования автоматически переведенных субтитров с YouTube. Язык, на который переводятся субтитры, выбирается согласно языку интерфейса расширения. Если язык интерфейса и язык автосгенерированных субтитров совпадают, то субтитры не будут показаны (#537)
- Добавлена возможность выбрать формат загружаемых субтитров
- Добавлено округление до целого процента в слайдере "Уменьшить громкость видео до"
- Добавлена возможность включить загрузку аудио дорожки и субтитров с названием видео, если расширение его нашло (#791)
- Добавлена возможность включить обход Media CSP через расширение (включено по умолчанию, если браузер поддерживает WebAudio), а не удалять его полностью через Tampermonkey или какие-либо другие расширения, что является не очень безопасным вариантом. Работает, только, для ниже перечисленных веб-сайтов. Если вы нашли сайт, который необходимо добавить в список для обхода Media CSP - создайте Issues.

  - "linkedin",
  - "sap",
  - "apple_developer",
  - "xvideos",
  - "twitter",
  - "facebook",
  - "yandexdisk",
  - "reddit",
  - "invidious",
  - "piped"

- Добавлено автоопределение возвращаемого типа данных из LocalStorage, если присутствует стандартное значение
- Убрано выставление языка `auto` по умолчанию для Invidious и Piped
- Автосгенериванные субтитры с YouTube, теперь, используют уже существующие токены, а не генерируют новые
- В субтитрах от самого веб-сайта, теперь, показывается полный домен сайта, а не его внутреннее название
- Изменена логика работы субтитров на "кастомных" сайтах. Теперь, вместо ошибки, возвращаются пустые данные
- Изменен загрузчик стилей при сборке расширения, благодаря этому скорость сборки стала немного быстрее, а итоговый размер кода, отвечающего за стили, уменьшен в ~1.65 раза
- Стандартный адрес для проксирования m3u8 изменен на `media-proxy.toil.cc/v1/proxy/m3u8`. Если вы развернули свой собственный m3u8 прокси вы можете перейти на более производительный прокси-сервера [FOSWLY/media-proxy](https://github.com/FOSWLY/media-proxy) или остаться на старой версии [FOSWLY/m3u8-proxy-worker](https://github.com/FOSWLY/m3u8-proxy-worker)
- Адрес vot-backend изменен на `https://vot.toil.cc/v1`
- Исправлена работа расширения без наличия WebAudio (#749)
- Исправлена ошибка из-за которой кнопка перевода могла не появляться до первичного получения субтитров или завершения автоперевода
- Исправлена ошибка из-за которой при смене адреса worker прокси-сервера домен сразу применялся, даже, если проксирование аудио выключено
- Исправлена ошибка из-за которой при запросе субтитров не учитывался выбранный язык видео
- Исправлена ошибка из-за которой для определения языка видео на ютубе использовалось локализованное название видео
- Исправлена ошибка из-за которой субтитры на некоторых сайтах могли иметь неверный размер шрифта
- Исправлена ошибка из-за которой не отключалась озвучка после перелистывания видео в YT Shorts (#802)
- Исправлена работа расширения для Kick
- Исправлена работа расширения для Coursera (#823)
- Исправлено появление значка загрузки при переводе на "кастомных" сайтах
- Исправлено появление кнопки в Rumble (#784)
- Исправлен жирный размер текста в меню и кнопке перевода на некоторых сайтах
- Исправлена растянутая кнопка перевода в Safari (#497)
- Исправлена попытка активации расширения на accounts.youtube.com
- Автоперевод и первичное получение субтитров, теперь, происходит параллельно
- Некоторые другие мелкие улучшения

## What's Changed
* Cleanup code by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/757
* Hotkey to enable/disable translation by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/760
* Rework l10n by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/797
* Added some locale info by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/798
* Added sap learning by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/801
* Fix YT shorts by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/803
* Upgrade to votjs 1.2.9 by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/812
* Bypass Media CSP by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/813
* fix wiki-gen script by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/814
* Fix yarn by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/817
* Add id3 to audio by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/820
* Fix coursera  by @ilyhalight in https://github.com/ilyhalight/voice-over-translation/pull/825

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.6.1...1.7.0

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.7.0'>1.7.0</a>.</em>

**Comment by FiLinX** [comment]
я так понимаю, что в связи новой функцией "обход Media CSP через расширение" - мне можно в настройках темпермоки изменить "Добавить TM в политику безопасности CSP:" с Да на Нет  ?

**Reply by ilyhalight** [reply]
Лучше поставить на автоматически, особенно, если используемый браузер firefox

**Comment by hrihoriart** [comment]
спасибо разработчикам этого расширения, оно улучшило мой процесс обучения в разы!

**Comment by DanDiablo83** [comment]
У меня не получилось запустить его в браузере из трех букв (Бог еще такой был и молот у него), хотя Фокс поддерживается. Скрипт установился, но кнопки Перевести не появилось. Скрипт не поддерживает данный браузер? Можно ли это исправить?

**Reply by ilyhalight** [reply]
Есть какие-то ошибки в консоли?

**Reply by DanDiablo83** [reply]
Кстати, есть, как то я затупил и туда не посмотрел даже.

<img width="678" alt="Screenshot_2" src="https://github.com/user-attachments/assets/8d97e872-3976-49c5-a1b0-d86087d7b0ee">

**Reply by ilyhalight** [reply]
> Кстати, есть, как то я затупил и туда не посмотрел даже.
> <img alt="Screenshot_2" width="678" src="https://private-user-images.githubusercontent.com/60819326/371860992-8d97e872-3976-49c5-a1b0-d86087d7b0ee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc2MjI2ODgsIm5iZiI6MTcyNzYyMjM4OCwicGF0aCI6Ii82MDgxOTMyNi8zNzE4NjA5OTItOGQ5N2U4NzItMzk3Ni00OWM1LWExYjAtZDg2MDg3ZDdiMGVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI5VDE1MDYyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM0NDc4MWM3YTRmOWZhNTdjMGY5OWRhMTA5NzM4YTIwNGFhMmRjYzY4ZWNmMWQ1NTU4ZTkwNGI5YTcwZmEyNGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.lwUp8DzOdCkcYH1eay4mjkDP77GfnOsrevCMoQbKVcY">

Вроде исправил, обновитесь через кнопку "Установить скрипт"

**Reply by DanDiablo83** [reply]
Благодарю! Дай Бог здоровья, спасли малограмотных, как я.
